sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala

```diff
@@ -131,6 +131,24 @@ class DataFrameStatSuite extends QueryTest {
     items2.getSeq[Double](0) should contain (-1.0)
   }
 
+  test("Frequent Items 2") {
+    val rows = sqlCtx.sparkContext.parallelize(Seq.empty[Int], 4)
+    // this is a regression test, where when merging partitions, we omitted values with higher
+    // counts than those that existed in the map when the map was full. This test should also fail
+    // if anything like SPARK-9614 is observed once again
+    val df = rows.mapPartitionsWithIndex { (idx, iter) =>
+      if (idx == 3) {
+        Iterator.tabulate(5)(_ => "3")
+      } else {
+        Iterator.tabulate(5)(i => (i * (idx + 1)).toString)
+      }
+    }.toDF("a")
+    val results = df.stat.freqItems(Array("a"), 0.25)
+    val items = results.collect().head.getSeq[String](0)
+    items should contain ("3")
```

Comment:
minor: I checked `DataFrameSuite` and didn't see `should` usage. `assert(....contains(...))` is more common, just to be consistent with code in other places.